src/console/ConsoleReaderChildProcessDaemon.php

```diff
@@ -104,21 +101,44 @@ private function shutdownSubprocess() : void{
 		//the first place).
 		proc_terminate($this->subprocess);
 		proc_close($this->subprocess);
-		stream_socket_shutdown($this->socket, STREAM_SHUT_RDWR);
 	}
 
 	public function readLine() : ?string{
 		$r = [$this->socket];
 		$w = null;
 		$e = null;
 		if(stream_select($r, $w, $e, 0, 0) === 1){
-			$command = fgets($this->socket);
-			if($command === false){
+			$line = fgets($this->socket);
+			if($line === false){
 				$this->logger->debug("Lost connection to subprocess, restarting (maybe the child process was killed from outside?)");
 				$this->shutdownSubprocess();
 				$this->prepareSubprocess();
 				return null;
 			}
+			$line = rtrim($line, "\n");
+
+			if($line === ""){
+				//keepalive
+				return null;
+			}
+
+			$command = null;
+
+			$parts = explode(":", $line, 2);
```

Comment:
This won't work correctly if the command contains a : character. Noticed this when I flipped the params to begin with, but forgot to fix it.